### DISCUSSION TITLE
Possible bug for findOneAndUpdate set field how generic object

### DISCUSSION THREAD

**Discussion by cesxhin** [Automattic/mongoose#14980] [Q&A]
A created this schema

`
Schema({
    test: Object
}, {minimize: false});
`

and when I findOneAndUpdate the fields are always reset rather than added.

await model.findOneAndUpdate({}, {$set: {test: { cat: "mew!" }}}, {new: true}) //show -> cat

await model.findOneAndUpdate({}, {$set: {test: { dog: "bau!" }}}, {new: true}) //show only -> dog (the cat should also show)

**Comment by vkarpov15** [comment]
You would do `$set: { 'test.dog': 'bau!' }` to add a `dog` field to `test` without overwriting the existing `test` property. When you do `$set: { test: { dog: 'bau' } }`, the MongoDB server interprets that as "replace the entire `test` property with the object `{ dog: 'bau' }`
